### DISCUSSION TITLE
[DISCUSS] Add a Python interface

### DISCUSSION THREAD

**Discussion by MrPowers** [unitycatalog/unitycatalog#213] [Ideas]
A Python interface would make this project more approachable to Python devs.

It's relatively easy to add an initial Python interface with [Stainless](https://www.stainlessapi.com/), but customizations would be needed of course.

Here's [an example repo](https://github.com/samster25/unitycatalog-python) that's built with Stainless.

Let me know what you think!

**Comment by Fokko** [comment]
Thanks for bringing this up! I've been using PyIceberg to talk with the Iceberg REST interface:

![image](https://github.com/unitycatalog/unitycatalog/assets/1134248/f6af5f73-79af-414d-acaf-70652bb41b50)

Some things don't work yet (creating tables), but that will hopefully be resolved soon 👍

**Comment by MrPowers** [comment]
@Fokko - oh, that's cool 😎 

Here's [the video where the Daft team talks about the Python interface they created](https://www.youtube.com/live/8i5eOE3KdBo?t=458s).

Do you think we should also add a `pip install unitycatalog` like interface where devs can access this lib directly via Python?

**Comment by Fokko** [comment]
> Do you think we should also add a pip install unitycatalog like interface where devs can access this lib directly via Python?

For sure, I think there is room for another library that talks unity REST API. 👍

**Comment by jaychia** [comment]
I think it's healthiest to probably have:

1. SDK generated with Stainless (in lock-step with the OpenAPI spec)

2. A higher-level Python SDK if you envision deeper integrations with other tools such as Daft, Pandas, Polars etc. basically it would chain the API calls for higher level functionality such as reading tables into other frameworks.

**Comment by samster25** [comment]
> A higher-level Python SDK if you envision deeper integrations with other tools such as Daft, Pandas, Polars etc. basically it would chain the API calls for higher level functionality such as reading tables into other frameworks.

I believe we should also be able to do this within the same `unitycatalog` package, offering both the low level and high level apis for engine developers / end users respectively.

**Comment by djouallah** [comment]
why not Unity OSS adopt iceberg catalog API and everyone will be happy and we will get world peace ?
